src/commands/build-util.ts

```diff
@@ -121,10 +121,10 @@ export function wrapImportMeta({
   }
   return (
     (hmr
-      ? `import * as  __SNOWPACK_HMR__ from '/${buildOptions.metaDir}/hmr.js';\nimport.meta.hot = __SNOWPACK_HMR__.createHotContext(import.meta.url);\n`
+      ? `import * as  __SNOWPACK_HMR__ from '${buildOptions.baseUrl.replace(/\/$/, '')}/${buildOptions.metaDir}/hmr.js';\nimport.meta.hot = __SNOWPACK_HMR__.createHotContext(import.meta.url);\n`
       : ``) +
     (env
-      ? `import __SNOWPACK_ENV__ from '/${buildOptions.metaDir}/env.js';\nimport.meta.env = __SNOWPACK_ENV__;\n`
+      ? `import __SNOWPACK_ENV__ from '${buildOptions.baseUrl.replace(/\/$/, '')}/${buildOptions.metaDir}/env.js';\nimport.meta.env = __SNOWPACK_ENV__;\n`
```

Comment:
One suggestion since this shares a bit of logic: could you dedupe these into a helper function in this file? e.g.:

```ts
const getMetaFile = (config: SnowpackConfig, filename: string) => `${buildOptions.baseUrl.replace(/\/$/, '')}/${buildOptions.metaDir}/${filename}`;
```

Then do something like:

`import * as  __SNOWPACK_HMR__ from '${getMetaFile(config, 'hmr.js')}`

I know there’s a bit of mess in this file, but that would help us make later additions (for example, I’m not 100% sure we’re replacing Windows backslashes with forward slashes, and if we needed to update the `replace()` function it’d be easy to miss one if it’s duplicated.)